extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala

```diff
@@ -301,6 +318,20 @@ trait LineageParser {
       case p: LocalRelation =>
         joinRelationColumnLineage(parentColumnsLineage, p.output, Seq(LOCAL_TABLE_IDENTIFIER))
 
+      case p: InMemoryRelation if p.cacheBuilder.tableName.nonEmpty =>
+        val tableName = p.cacheBuilder.tableName.get
+        sparkSession.sessionState.catalog.getTempView(tableName) match {
```

Comment:
Is it possible to use `InMemoryRelation.cacheBuilder.cachedPlan.logicalLink` ? it covers more case and we can reuse the existed code to track lineage.

Reply:
Good idea, let me try it. Previously I never found out how to associate `sparkPlan` to `logicalPlan`

Reply:
`InMemoryRelation.cacheBuilder.cachedPlan.logicalLink` will return a `None` value in this case, it looks like `setLogicalLink` method is not called in the cache phase. This way does not work.

Reply:
I think it works. But there is a gap between executed plan and spark plan, e.g. wholestagecodgen does not have a logical link. What we need is the spark plan, so we can skip those nodes.